pkg/querier/querier.go

```diff
@@ -50,9 +50,10 @@ type Config struct {
 
 // RegisterFlags register flags.
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	cfg.Engine.RegisterFlagsWithPrefix("querier", f)
 	f.DurationVar(&cfg.TailMaxDuration, "querier.tail-max-duration", 1*time.Hour, "Limit the duration for which live tailing request would be served")
-	f.DurationVar(&cfg.QueryTimeout, "querier.query_timeout", 1*time.Minute, "Timeout when querying backends (ingesters or storage) during the execution of a query request")
-	f.DurationVar(&cfg.ExtraQueryDelay, "distributor.extra-query-delay", 0, "Time to wait before sending more than the minimum successful query requests.")
+	f.DurationVar(&cfg.QueryTimeout, "querier.query-timeout", 1*time.Minute, "Timeout when querying backends (ingesters or storage) during the execution of a query request")
+	f.DurationVar(&cfg.ExtraQueryDelay, "querier.extra-query-delay", 0, "Time to wait before sending more than the minimum successful query requests.")
```

Comment:
I'm also a bit concerned here about changing existing flags but i'm starting to think now maybe we should, the configs you are changing are unlikely to be used by anyone currently, especially at the command line.

Reply:
as mentioned above, adding a note on the PR itself after re-thinking this a bit

Reply:
Yes, it would be breaking change for anyone using cmd line flags. But I think it's better we change it now, I am afraid we may not revisit these changes again.